htdocs/comm/propal/list.php

```diff
@@ -1270,10 +1310,49 @@
 	$delallowed = $user->rights->propal->creer;
 
 	print $formfile->showdocuments('massfilesarea_proposals', '', $filedir, $urlsource, 0, $delallowed, '', 1, 1, 0, 48, 1, $param, $title, '', '', '', null, $hidegeneratedfilelistifempty);
-} else {
-	dol_print_error($db);
-}
 
-// End of page
-llxFooter();
-$db->close();
+	if ($action == 'validate') {
+	if (GETPOST('confirm') == 'yes') {
+		$tmpproposal = new Propal($db);
+		foreach ($toselect as $checked) {
+			$tmpproposal->fetch($checked);
+			if ($tmpproposal->fetch($checked)) {
+				if ($tmpproposal->statut == 0) {
+					$tmpproposal->valid($user);
```

Comment:
You must add a test here on result on $tmpproposal->valid($user);
to show a setEventMessage of error if it fails.